discovery-infra/tests/test_e2e_install.py

```diff
@@ -1,32 +1,42 @@
 import time
-from typing import Tuple
 
 import pytest
+from _pytest.fixtures import FixtureLookupError
 from junit_report import JunitTestSuite
+
 from test_infra.consts import OperatorStatus
 
 from tests.base_test import BaseTest
-from tests.config import ClusterConfig, TerraformConfig
+from tests.config import ClusterConfig
 from tests.conftest import get_available_openshift_versions, get_api_client
 
 
 class TestInstall(BaseTest):
 
+    @pytest.fixture
+    def cluster_configuration(self, request):
+        # Overriding the default BaseTest.cluster_configuration fixture to set the openshift version.
+        config = ClusterConfig()
+
+        try:
+            # Resolving the param value.
+            version = request.getfixturevalue("openshift_version")
+            config.openshift_version = version
+        except FixtureLookupError:
+            pass
+
+        return config
```

Comment:
There are lots of times that we would like to override the default configurations. I don't think we should specify each scenario in this fixture, hence I don't find it so beneficial.
IMO it can be removed.

Reply:
The fixtures in the base test supports multi-platfrom, it creates the right controller configuration object then creates the right controller object and initialized it properly.
The test's programmer shouldn't deal with the different platform creation logic.
If the test requires updating the configuration object it should override the cluster_configuration fixture and return the configuration object.
For a common cases we can extend the base fixture and add marks.